cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js

```diff
@@ -432,15 +433,24 @@ class HydratorPlusPlusNodeConfigCtrl {
       scope: this.$scope
     };
 
-    let connections = this.ConfigStore.getConfigForExport().config.connections;
+    let { stages, connections } = this.ConfigStore.getConfigForExport().config;
     let adjacencyMap = this.HydratorPlusPlusOrderingFactory.getAdjacencyMap(connections);
     let postBody = [];
-    angular.forEach(adjacencyMap, (value, key) => {
-      let inputPlugin = value.filter(node =>this.state.node.plugin.label === node);
-      if (inputPlugin.length) {
-        postBody.push(key);
+    let previousStageName = Object.keys(adjacencyMap).find(key => adjacencyMap[key].indexOf(this.state.node.plugin.label) !== -1);
+
+    if (previousStageName) {
+      let previousStage = stages.find(stage => stage.name === previousStageName);
+
+      // In case we have multiple condition nodes in a row, we have to keep traversing back
+      // until we find a node that actually has records out
+      while (previousStage && previousStage.plugin.type === 'condition') {
+        previousStageIsCondition = true;
+        previousStageName = Object.keys(adjacencyMap).find(key => adjacencyMap[key].indexOf(previousStageName) !== -1);
```

Comment:
This should be a complete check right? `adjacency[key] === previewStageName` .. `indexOf` might be a problem if we have `Condition1` and `Condition12`.

Reply:
`adjacency[key]` returns an array, and even if `adjacency[key] = ['Condition1', 'Condition12']`, the `indexOf` would still find the correct value.